cacao: use system's boehm garbage collector

This is a Gentoo's patch [1] to force the use of system's bdwgc.

Cacao was installing an internal version of bdwgc which conflicts
with the system's version.

[1] https://gitweb.gentoo.org/proj/java.git/plain/dev-java/cacao/files/system-boehm-gc.patch

Upstream-Status: Inappropriate [not author]

Signed-off-by: Hugo Vasconcelos Saldanha <hugo.saldanha@aker.com.br>

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -691,6 +691,8 @@
 
 AC_CHECK_ENABLE_PIC_ASM
 
+PKG_CHECK_MODULES([BOEHM], bdw-gc,,AC_MSG_ERROR[boehm-gc not found])
+
 dnl Define version numbers.
 AC_VERSION_DETAIL
 AC_VERSION_CONFIG
@@ -793,12 +795,6 @@
 )
 
 
-dnl We only build the shared library objects but we use it as
-dnl noinst_LTLIBRARIES.
-ac_configure_args="$ac_configure_args --disable-static"
-AC_CONFIG_SUBDIRS(src/mm/boehm-gc)
-
-
 AC_OUTPUT
 
 
diff --git a/src/mm/Makefile.am b/src/mm/Makefile.am
--- a/src/mm/Makefile.am
+++ b/src/mm/Makefile.am
@@ -35,14 +35,13 @@
 endif
 
 if ENABLE_GC_BOEHM
-SUBDIRS = \
-	boehm-gc
-
 GC_FILE = \
 	gc-boehm.cpp
 
-GC_LIB = \
-	$(top_builddir)/src/mm/boehm-gc/libgc.la
+AM_CPPFLAGS += $(BOEHM_CFLAGS)
+
+GC_LIB = $(BOEHM_LIBS)
+
 endif
 
 if ENABLE_GC_CACAO
